File: architecturecomponents/src/main/java/com/fesskiev/architecturecomponents/di/AppModule.java
package com.fesskiev.architecturecomponents.di;


import android.content.Context;

import com.fesskiev.architecturecomponents.utils.Utils;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private Context context;

    public AppModule(Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    Context provideApplication() {
        return context;
    }

    @Provides
    @Singleton
    Utils provideUtils(Context context) {
        return new Utils(context);
    }
}

File: kotlinsamples/src/main/java/com/fesskiev/kotlinsamples/di/AppModule.kt
package com.fesskiev.kotlinsamples.di

import android.app.Application
import android.content.Context
import dagger.Binds
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
abstract class AppModule {

    @Binds
    abstract fun provideContext(application: Application): Context
}